packages/core/helper-plugin/src/components/SearchURLQuery.tsx

```diff
@@ -1,28 +1,39 @@
-import React, { useLayoutEffect, useRef, useState } from 'react';
+import * as React from 'react';
 
 import { Icon, IconButton, Searchbar, SearchForm } from '@strapi/design-system';
 import { Search as SearchIcon } from '@strapi/icons';
-import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
 
-import { useTracking } from '../features/Tracking';
+import { TrackingEvent, useTracking } from '../features/Tracking';
 import { useQueryParams } from '../hooks/useQueryParams';
 
-const SearchURLQuery = ({ label, placeholder, trackedEvent, trackedEventDetails }) => {
-  const inputRef = useRef(null);
-  const iconButtonRef = useRef(null);
+export interface SearchURLQueryProps {
+  label: string;
+  placeholder?: string;
+  trackedEvent?: TrackingEvent['name'] | null;
+  trackedEventDetails?: TrackingEvent['properties'];
+}
+
+const SearchURLQuery = ({
+  label,
+  placeholder = undefined,
+  trackedEvent = null,
+  trackedEventDetails = undefined,
+}: SearchURLQueryProps) => {
+  const inputRef = React.useRef<HTMLInputElement>(null);
+  const iconButtonRef = React.useRef<HTMLButtonElement>(null);
 
   const [{ query }, setQuery] = useQueryParams();
 
-  const [value, setValue] = useState(query?._q || '');
-  const [isOpen, setIsOpen] = useState(!!value);
+  const [value, setValue] = React.useState((query?._q as string) || '');
```

Comment:
How come you have to cast this to `string`?

Reply:
Because otherwise I have this error
<img width="886" alt="cast" src="https://github.com/strapi/strapi/assets/2589748/5f7e432f-fd96-4fec-8139-239eb967481d">

Reply:
that's not really a good reason to do that change then, we probably need to understand which type is right / wrong 🤔

Reply:
and also
![Uploading other error.png…]()

Reply:
removed the cast and added type checking